crates/fluvio/src/fluvio.rs

```diff
@@ -102,19 +102,11 @@ impl Fluvio {
         Self::connect_with_config(&config).await
     }
 
-    /// Creates a new Fluvio client with the given connector and configuration
-    pub async fn connect_with_connector(
-        connector: DomainConnector,
+    /// Creates a new Fluvio client with the given configurations
+    pub async fn connect_with_client_config(
+        client_config: ClientConfig,
```

Comment:
Current terminology is confusing with many config.  Let's clean up this.
`FluvioConfig` should be renamed `FluvioClusterConfig`.

Then this would be:
```
pub async fn connect_with_client_config(
         client_config: ClientConfig,
         fluivo_config: &FluvioClusterconfig
}
```

Why do we need to pass `ClientConfig`?

Reply:
make private

Reply:
ok, but that will change 40 files

Reply:
> Why do we need to pass ClientConfig?
> Should not expose it

to use it on retry, so pub(crate) then?

Reply:
Yes, at least

Reply:
done, renamed to FluvioClusterconfig and connect_with_client_config is pub(crate) now